app/rdf/query-hierarchies.ts

```diff
@@ -95,17 +135,45 @@ export const queryHierarchy = async (
     return null;
   }
 
-  const dimensionValuesProm = queryDimensionValues(dimensionIri, sparqlClient);
+  const dimensionValuesWithLabelsProm = queryDimensionValuesWithLabels({
+    dimensionIri,
+    client: sparqlClient,
+    locale,
+  });
   const results = await getHierarchy(hierarchy).execute(
     // @ts-ignore
     sparqlClientStream,
     rdf
   );
 
   const tree = toTree(results, dimensionIri, locale);
-  const dimensionValues = new Set(await dimensionValuesProm);
-  return mapTree(
-    pruneTree(tree, (node) => dimensionValues.has(node.value)),
-    (node) => ({ ...node, hasValue: dimensionValues.has(node.value) })
+  const treeValues = getOptionsFromTree(tree).map((d) => d.value);
+  const dimensionValuesWithLabels = await dimensionValuesWithLabelsProm;
+  const dimensionValues = dimensionValuesWithLabels.map((d) => d.value);
+  const prunedTree = mapTree(
+    pruneTree(tree, (node) => dimensionValues.includes(node.value)),
+    (node) => ({
+      ...node,
+      hasValue: dimensionValues.includes(node.value),
+    })
+  );
+  const additionalTreeValues = dimensionValuesWithLabels
+    .filter((d) => !treeValues.includes(d.value))
+    .map((d) => ({
+      label: d.label || "-",
+      value: d.value,
+      depth: -1,
+      children: [],
+      dimensionIri,
+      hasValue: true,
+    }));
+
+  return sortTree([...prunedTree, ...additionalTreeValues], (a, b) =>
+    a.depth === -1 && b.depth === -1
+      ? defaultSort(a, b)
+      : a.depth === -1 || b.depth === -1
```

Comment:
I do not think it works correctly since you check both a and b. Here I think only `a` should be checked.

```
> // INCORRECT
> [{d:-1, a:'hey'}, {d:1}, { d: -1 }].sort((a, b) => a.d === -1 || b.d === -1 ? 1 : 0)
Array(3) [ {…}, {…}, {…} ]
0: Object { d: -1 }
1: Object { d: 1 }
2: Object { d: -1, a: "hey" }
> // CORRECT
> [{d:-1, a:'hey'}, {d:1}, { d: -1 }].sort((a, b) => a.d === -1 ? 1 : 0)
> Array(3) [ {…}, {…}, {…} ]
0: Object { d: 1 }
1: Object { d: -1 }
2: Object { d: -1, a: "hey" }
```

Reply:
Yes, I don't know how I created this 😅 I switched to using `d3.ascending` & `d3.descending` here.